honeybadgermpc/polynomial.py

```diff
@@ -94,17 +111,15 @@ def get_omega(field, n, seed=None):
 
     This only makes sense if n is a power of 2!
     """
-    p = field.modulus
+    assert n & n-1 == 0, "n must be a power of 2"
     if seed is not None:
         random.seed(seed)
-    x = field(random.randint(0, p-1))
-    y = pow(x, (p-1)//n)
-    if y == 1:
+    x = field(random.randint(0, field.modulus-1))
+    y = pow(x, (field.modulus-1)//n)
+    if y == 1 or pow(y, n//2) == 1:
         return get_omega(field, n)
-    if pow(y, n//2) == 1:
-        return get_omega(field, n)
-    assert pow(y, n) == 1
-    assert pow(y, n//2) != 1
+    assert pow(y, n) == 1, "omega must be 2n'th root of unity"
+    assert pow(y, n//2) != 1, "omega must be primitive 2n'th root of unity"
```

Comment:
Same comment as in https://github.com/initc3/HoneyBadgerMPC/pull/51/files#r216349389